Measurement tools and minors patches

Widelx opened the pull request:
Stable version until [commit](https://github.com/BatoSEC25-ECN/girouette/pull/6/commits/7ba09060c27cb374f8c260dcb892c56197aeeba).

It might happen that I push new modifications after this commit but I must test those to ensure non regression.
Be aware of that when reviewing.

Moreover, to me `measure tool` folder should be placed inside another directory. However having it in `code` makes sense : for examples `dependencies` folder and `setup.adoc` should cover this tools, IMO. 
I'm down for other ideas so we can choose if we keep it here or move it elsewhere.

Widelx committed:
doc: add fusion360 download page

Widelx committed:
chore: update gitignore and required tools

Widelx committed:
doc: add keysight developper datasheet

Widelx committed:
doc: improve setup documentation

Widelx committed:
feat!: create MCU pwm burst generation

Widelx committed:
feat!: add keysight custom library and required dataclasses

Widelx committed:
feat: add a logger for debug purpose

Widelx committed:
feat: create a plotter to compare two data curve

Widelx committed:
chore: add python libraries requirements

Widelx committed:
feat!: create main runner script

Widelx committed:
fix: remove uncrustify config (duplicated, wrong folder)

Widelx committed:
refactor: remove unused dependencies

K40NN commented on code/measure_tool/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+pandas==2.2.3
+plotly==6.0.1
+PyVISA==1.14.1
```
seem's there a  missing in the requirement; on my side project isn't compiling

Widelx replied:
reran `pipreqs`, no update found. Might be on your side.

cf https://stackoverflow.com/questions/35796968/get-all-modules-packages-used-by-a-python-project

Widelx committed:
feat!: add possibility of running multiple tests

Add multiple features to run "infinite" amount of tests:
 - plotter handles an undefined amount of .csv files from a list (restr
   icted to 5 colors though, curves color could overlap)
 - keysight file formatting is standardized (lower case, ensure .csv at
   the end, replace spaces with underscore)
 - main is updated to run an undefined amount of measurement, prompting
   the user to run the next test

Widelx committed:
refactor(dataclasses): improve readability and split some functions logic

Widelx committed:
refactor(keysight): improve readability and split some functions logic

Widelx committed:
refactor(plotter): improve readability and split some functions logic

Widelx committed:
refactor: update main to match previous changes

K40NN commented on code/code_test/main/code_test.c:
```diff
@@ -1,7 +1,76 @@
+    ESP_ERROR_CHECK(ledc_set_duty(LEDC_MODE, LEDC_CHANNEL, LEDC_DUTY));
+    ESP_ERROR_CHECK(ledc_update_duty(LEDC_MODE, LEDC_CHANNEL));
+    esp_rom_delay_us(BURST_DELAY_MICRO);
+    ESP_ERROR_CHECK(ledc_stop(LEDC_MODE, LEDC_CHANNEL, 0));
```
the famous macro error.

K40NN commented on code/measure_tool/keysight.py:
```diff
@@ -0,0 +1,240 @@
+from decimal import Decimal
+
+import csv
+import pyvisa
```
it's seems i encounter issue on my side  with pyvisa. i will investigate this point .

Widelx replied:
IMO it's on your side. It runs well for me 😄

K40NN commented on code/measure_tool/logger.py:
```diff
@@ -0,0 +1,44 @@
+import logging
```
smart habbit

Widelx replied:
Yeah, it's quite useful and powerful. 
Makes more sense than confusing prints !

K40NN commented on code/setup.adoc:
Suite Main

K40NN commented on code/tools/uncrustify.cfg:
```diff
@@ -481,4 +481,4 @@ sp_inside_tparen                        = remove        # string (add/force/igno
 # Add or remove space between nested parentheses, i.e. '((' vs. ') )'.
-sp_paren_paren                          = remove   # ignore/add/remove/force/not_defined
\ No newline at end of file
+sp_paren_paren                          = remove   # ignore/add/remove/force/not_definedez
```
not_definedez ? not defineded seems better

K40NN commented on code/tools/uncrustify.exe:
binary in a commit is it normal ? maybe this file can be added on git ignore

Widelx replied:
IMO it makes sense, `libusb.exe` is embedded as well. 
Those executables are added to simplify user install process. 
Those tools can be quite tedious to find 😸

K40NN commented on modeling/setup.adoc:
```diff
@@ -3,7 +3,7 @@ Kadwalon Lobet, Guillaume de Maricourt.
 This project uses fusion 360 as CAD software. +
 To setup this tool, enable link:https://www.autodesk.com/fr/products/fusion-360/education[fusion 360 school licence]. +
-Follow the installation process.
+You can then install fusion from this lin:https://manage.autodesk.com/products/all[page].
```
fusion finally get back the free utilisation for individual uses and maker. but all features is provided , i recomand the student licences witch can be obtain via shool @.

Widelx replied:
Did not know, thanks !

K40NN reviewed (commented):
nothing important , mostly comment about my side or smart stuff you did. exept one typo all is great

Widelx committed:
fix:clean probe ratio

D:\Documents\41_Ecole\4_Centrale_Nantes\A3\_Projet_bateau\girouette\code\measure_tool\main.py

Widelx committed:
refactor(dataclasses): refactor code

D:\Documents\41_Ecole\4_Centrale_Nantes\A3\_Projet_bateau\girouette\code\measure_tool\dataclass.py

Widelx committed:
feat: reimplement probe ratio

D:\Documents\41_Ecole\4_Centrale_Nantes\A3\_Projet_bateau\girouette\code\measure_tool\dataclass.pyD:\Documents\41_Ecole\4_Centrale_Nantes\A3\_Projet_bateau\girouette\code\measure_tool\keysight.py

Widelx committed:
refactor(keysight): refactor code

Widelx committed:
fix: fix plotter issues

D:\Documents\41_Ecole\4_Centrale_Nantes\A3\_Projet_bateau\girouette\code\measure_tool\plotter.py

Widelx committed:
fix: patch frequency clamping

Widelx committed:
feat: add horizontal markers

Widelx committed:
close project
